db/view/view.cc

```diff
@@ -2693,12 +2693,94 @@ static future<> flush_base(lw_shared_ptr<replica::column_family> base, abort_sou
     }).discard_result();
 }
 
+future<> view_builder::update_base_schema(const replica::column_family& base) {
+    auto base_schema = base.schema();
+    auto& views = base.views();
+    if (views.empty()) {
+        co_return;
+    }
+    auto step_it = _base_to_build_step.find(base_schema->id());
+    if (step_it == _base_to_build_step.end()) {
+        co_return;// In case all the views for this CF have finished building already.
+    }
+    if (step_it->second.base_schema->version() == base_schema->version()) {
+        co_return;// Base schema was already updated, probably while creating an index.
+    }
+    step_it->second.base_schema = base_schema;
+    // Update view_ptrs in all build statuses with new view_ptr from the base table.
+    // If some were dropped, stop them as well, so we won't continue building them,
+    // especially as their schema is not updated with the new base schema.
+    // If some were added, we can handle them later, in on_create_view.
+    std::vector<view_ptr> views_to_remove;
+    for (auto& status : step_it->second.build_status) {
+        auto view_it = std::ranges::find_if(views, [&status] (const view_ptr& v) {
+            return v->id() == status.view->id();
+        });
+        if (view_it == views.end()) {
+            views_to_remove.push_back(status.view);
+        } else {
+            // This update will be performed again in on_update_view, but it will set the same
+            // view_ptr, and it's still needed there because not all view schema updates are
+            // caused by base schema updates.
+            status.view = *view_it;
+        }
+    }
+    for (auto& view : views_to_remove) {
+        co_await drop_view(base_schema->ks_name(), view->cf_name(), step_it->second);
+    }
+}
+
+future<> view_builder::drop_view(const sstring& ks_name, const sstring& view_name, build_step& step) {
+    // The view is absent from the database at this point, so find it by brute force.
```

Comment:
What if you DROP MATERIALIZED VIEW and then CREATE MATERIALIZED VIEW with exactly the same name?
I guess that we're fine because we only call this new drop_view function when we know that the base no longer has **any** view with this name, so it's safe to stop any onegoing build with that name. But maybe a comment is needed to explain this or something - the function name "drop_view()" sounds like it might be used in other cases as well.